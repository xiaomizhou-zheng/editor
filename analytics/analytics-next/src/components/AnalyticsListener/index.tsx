import LegacyAnalyticsListener from './LegacyAnalyticsListener';
import ModernAnalyticsListener from './ModernAnalyticsListener';
import { AnalyticsListenerFunction } from './types';

let ExportedAnalyticsListener: AnalyticsListenerFunction;
if (
  typeof process !== 'undefined' &&
  process !== null &&
  process.env?.['ANALYTICS_NEXT_MODERN_CONTEXT']
) {
  ExportedAnalyticsListener = ModernAnalyticsListener as any;
} else {
  ExportedAnalyticsListener = LegacyAnalyticsListener as any;
}

export default ExportedAnalyticsListener;
