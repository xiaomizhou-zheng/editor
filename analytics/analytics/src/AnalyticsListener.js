/* eslint-disable react/sort-comp */
import React, { Component } from 'react';
import PropTypes from 'prop-types';

import matchEvent from './matchEvent';

/*
The Listener component is responsible for calling its `onEvent` handler when a
child component fires an analytics event, and passing the event up the hierarchy
*/
const ContextTypes = {
  onAnalyticsEvent: PropTypes.func,
};

if (process.env.NODE_ENV !== 'production' && !process.env.CI) {
  // eslint-disable-next-line no-console
  console.warn(
    'The @atlaskit/analytics package has been deprecated. Please use the @atlaskit/analytics-next package instead.',
  );
}

class AnalyticsListener extends Component {
  static defaultProps = {
    match: '*',
    matchPrivate: false,
  };

  static contextTypes = ContextTypes;

  static childContextTypes = ContextTypes;

  getChildContext() {
    return {
      onAnalyticsEvent: this.onAnalyticsEvent,
    };
  }

  onAnalyticsEvent = (name, data, isPrivate) => {
    // Call this component's onEvent method if it's a match
    // eslint-disable-next-line react/prop-types
    const { onEvent, match, matchPrivate } = this.props;
    if (
      matchPrivate === isPrivate &&
      matchEvent(match, name) &&
      typeof onEvent === 'function'
    ) {
      // send a clean data object so it can't be mutated between listeners
      const eventData = { ...data };
      onEvent(name, eventData);
    }

    // Pass the event up the hierarchy
    const { onAnalyticsEvent } = this.context;
    if (typeof onAnalyticsEvent === 'function') {
      onAnalyticsEvent(name, data, isPrivate);
    }
  };

  render() {
    const { children } = this.props; // eslint-disable-line react/prop-types
    return React.Children.only(children);
  }
}

export default AnalyticsListener;
