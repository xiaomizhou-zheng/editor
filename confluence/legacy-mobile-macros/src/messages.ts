import { defineMessages } from 'react-intl-next';

import {
  ERROR_LOADING_TEXT,
  FINAL_ERROR_LOADING_TEXT,
  TAP_TO_LOAD_TEXT,
  TAP_TO_REFRESH_PAGE_TEXT,
  TAP_TO_RETRY_TEXT,
  TAP_TO_VIEW_TEXT,
} from './ui/MacroComponent/MacroFallbackComponent/constants';

export const legacyMobileMacrosMessages = defineMessages({
  tapToLoadMacro: {
    id: 'fabric.editor.confluence.legacyMobileMacros.tapToLoadMacro',
    defaultMessage: TAP_TO_LOAD_TEXT,
    description: 'Tap block to load macro',
  },
  tapToViewMacro: {
    id: 'fabric.editor.confluence.legacyMobileMacros.tapToViewMacroMacro',
    defaultMessage: TAP_TO_VIEW_TEXT,
    description: 'Tap block to view macro',
  },
  tapToRetryLoadingMacro: {
    id: 'fabric.editor.confluence.legacyMobileMacros.tapToRetryLoadingMacro',
    defaultMessage: TAP_TO_RETRY_TEXT,
    description: 'Tap to retry loading macro after error',
  },
  errorLoadingMacro: {
    id: 'fabric.editor.confluence.legacyMobileMacros.errorLoadingMacro',
    defaultMessage: ERROR_LOADING_TEXT,
    description: 'Error loading macro',
  },
  finalErrorLoadingMacro: {
    id: 'fabric.editor.confluence.legacyMobileMacros.finalErrorLoadingMacro',
    defaultMessage: FINAL_ERROR_LOADING_TEXT,
    description: 'Could not load macro after final retry',
  },
  tapToRefreshPage: {
    id: 'fabric.editor.confluence.legacyMobileMacros.tapToRefreshPage',
    defaultMessage: TAP_TO_REFRESH_PAGE_TEXT,
    description: 'Tap block to refresh the macro view page',
  },
});
