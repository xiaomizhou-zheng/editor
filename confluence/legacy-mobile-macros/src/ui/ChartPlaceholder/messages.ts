import { defineMessages } from 'react-intl-next';

export const chartPlaceholderMessages = defineMessages({
  chartPlaceholderText: {
    id: 'fabric.editor.confluence.legacyMobileMacros.chartPlaceholderText',
    defaultMessage:
      'A chart should be here, but it canâ€™t be shown. Open this page in a browser to see the chart.',
    description:
      'Message shown in place of chart extension on mobile (where the extension is not yet supported)',
  },
});
