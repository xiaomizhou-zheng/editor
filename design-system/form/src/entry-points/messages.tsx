export { HelperMessage, ErrorMessage, ValidMessage } from '../messages';
