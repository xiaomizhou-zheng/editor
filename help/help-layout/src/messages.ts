import { defineMessages } from 'react-intl-next';

export const messages = defineMessages({
  help_loading: {
    id: 'helpPanel.loading',
    defaultMessage: 'Loading',
    description: '',
  },
  help_panel_header_title: {
    id: 'helpPanel.header.title',
    defaultMessage: 'Help',
    description: '',
  },
  help_panel_header_back: {
    id: 'helpPanel.header.back',
    defaultMessage: 'Back',
    description: '',
  },
  help_panel_header_close: {
    id: 'helpPanel.header.close',
    defaultMessage: 'Close',
    description: '',
  },
});
