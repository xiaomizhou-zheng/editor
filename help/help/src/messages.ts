import { defineMessages } from 'react-intl-next';

export const messages = defineMessages({
  help_loading: {
    id: 'help.loading',
    defaultMessage: 'Loading',
    description: 'Loading text',
  },
  help_header: {
    id: 'help.header',
    defaultMessage: 'Help',
    description: 'Header text',
  },
  help_header_whats_new: {
    id: 'help.header_whats_new',
    defaultMessage: `What's new`,
    description: `Header text in "What's New" mode`,
  },
  help_close: {
    id: 'help.close',
    defaultMessage: 'Close',
    description: 'Close text',
  },
  help_navigation_back: {
    id: 'help.navigation.back',
    defaultMessage: 'Back',
    description: 'Back text',
  },
  help_search_placeholder: {
    id: 'help.search.placeholder',
    defaultMessage: 'Search help articles',
    description: 'Place holder text for search input',
  },
  help_article_rating_title: {
    id: 'help.article_rating.title',
    defaultMessage: 'Was this helpful?',
    description: 'Rating article title text',
  },
  help_article_rating_option_yes: {
    id: 'help.article_rating.option.yes',
    defaultMessage: 'Yes',
    description: '"Yes" text used in the article rating options',
  },
  help_article_rating_option_no: {
    id: 'help.article_rating.option.no',
    defaultMessage: 'No',
    description: '"No" text used in the article rating options',
  },
  help_article_rating_form_title: {
    id: 'help.article_rating.form.title',
    defaultMessage: 'Tell us why you gave this rating',
    description: 'Title for the article rating form',
  },
  help_article_rating_accurate: {
    id: 'help.article_rating.accurate',
    defaultMessage: `It isn't accurate`,
    description: `Text for the "isn't accurate" option of the article rating form`,
  },
  help_article_rating_clear: {
    id: 'help.article_rating.clear',
    defaultMessage: `It isn't clear`,
    description: `Text for the "isn't clear" option of the article rating form`,
  },
  help_article_rating_relevant: {
    id: 'help.article_rating.relevant',
    defaultMessage: `It isn't relevant`,
    description: `Text for the "isn't relevant" option of the article rating form`,
  },
  help_article_rating_form_submit: {
    id: 'help.article_rating.form.submit',
    defaultMessage: `Submit`,
    description: `Text for the submit button in the article rating form`,
  },
  help_article_rating_form_Success: {
    id: 'help.article_rating.form.success',
    defaultMessage: `Thanks!`,
    description: 'Success message for the article rating form',
  },
  help_article_rating_form_failed: {
    id: 'help.article_rating.form.failed',
    defaultMessage: `We couldn't submit your feedback.`,
    description: 'Error message for the article rating form',
  },
  help_article_rating_form_failed_try_again: {
    id: 'help.article_rating.form.failed.try_again',
    defaultMessage: `Try again`,
    description: `Text for the "Try again" button of the article rating form`,
  },
  help_article_rating_form_cancel: {
    id: 'help.article_rating.form.cancel',
    defaultMessage: `Cancel`,
    description: `Text for the cancel button in the article rating form`,
  },
  help_article_rating_form_contact_me: {
    id: 'help.article_rating.form.contact_me',
    defaultMessage: `Atlassian can contact me about this feedback`,
    description:
      'Text for the "Contact me" checkbox in the article rating form',
  },
  help_article_list_item_type_whats_new: {
    id: 'help.article_list_item.type.whats_new',
    defaultMessage: `WHAT'S NEW`,
    description: `"What's new" article type title`,
  },
  help_article_list_item_type_help_article: {
    id: 'help.article_list_item.type.help-article',
    defaultMessage: `HELP ARTICLE`,
    description: 'Help article article type title',
  },
  help_related_article_title: {
    id: 'help.related_article.title',
    defaultMessage: 'Related articles',
    description: 'Title of the related articles list',
  },
  help_related_article_endpoint_error_title: {
    id: 'help.related_article.endpoint-error.title',
    defaultMessage: `We're having some trouble`,
    description:
      'Title for the message displayed when the related articles fetching fails',
  },
  help_related_article_endpoint_error_description: {
    id: 'help.related_article.endpoint-error.description',
    defaultMessage: `It's taking us longer than expected to show this content. It's probably a temporary problem.`,
    description:
      'Body for the message displayed when the related articles fetching fails',
  },
  help_related_article_endpoint_error_button_label: {
    id: 'help.related_article.endpoint-error.button-label',
    defaultMessage: `Try Again`,
    description:
      'text for the retry button displayed when the related articles fetching fails',
  },
  help_search_results_search_external_site: {
    id: 'help.search_results.search_external_site',
    defaultMessage: `Can't find what you're looking for? Try again with a different term or `,
    description:
      'Text for the button displayed at the end of the search result used to open an external search site',
  },
  help_search_results_no_results: {
    id: 'help.search_results.no_results',
    defaultMessage: `We can't find anything matching your search`,
    description: `Message displayed when the search doesn't return any results (line 1)`,
  },
  help_search_results_no_results_line_two: {
    id: 'help.search_results.no_results_line_two',
    defaultMessage: `Try again with a different term or `,
    description: `Message displayed when the search doesn't return any results (line 2)`,
  },
  help_search_results_external_site_link: {
    id: 'help.search_results.external_site_link',
    defaultMessage: `search all online help articles.`,
    description: `Text for the button used to open an external search site`,
  },
  help_article_error_title: {
    id: 'help.article_error.title',
    defaultMessage: `Something's gone wrong`,
    description: `Error message title for article loading`,
  },
  help_article_error_text: {
    id: 'help.article_error.text',
    defaultMessage: `Something prevented the page from loading. We will try to reconnect automatically.`,
    description: `Error message body for article loading`,
  },
  help_article_error_button_label: {
    id: 'help.article_error.button_label',
    defaultMessage: `Try now`,
    description:
      'text for the retry button displayed when an article fetching fails',
  },
  help_search_error: {
    id: 'help.search_error',
    defaultMessage: `We are having some trouble`,
    description: `Error message for article search loading (line 1)`,
  },
  help_search_error_line_two: {
    id: 'help.search_error_line_two',
    defaultMessage: `It's taking us longer than expected to show this content. It's probably a temporary problem.`,
    description: `Error message for article search loading (line 2)`,
  },
  help_search_error_button_label: {
    id: 'help.search_error.button_label',
    defaultMessage: `Retry`,
    description: 'text for the retry button displayed when the search fails',
  },
  help_whats_new_button_label: {
    id: 'help.whats_new.button_label',
    defaultMessage: 'Find out what’s changed in {productName}',
    description: `text for the "What's New" button`,
  },
  help_whats_new_button_label_without_product_name: {
    id: 'help.whats_new.button_label',
    defaultMessage: 'Find out what’s changed',
    description: `text for the "What's New" button`,
  },
  help_whats_new_button_tooltip: {
    id: 'help.whats_new.button_tooltip',
    defaultMessage: `See our latest feature announcements, improvements, bug fixes, and more.`,
    description: `text for the "What's New" tooltip`,
  },
  help_whats_new_filter_select_option_all: {
    id: 'help.whats_new.filter_select_option_all',
    defaultMessage: `All release types`,
    description: `text for the "Release types" dropdown`,
  },
  help_whats_new_filter_select_option_new: {
    id: 'help.whats_new.filter_select_option_new',
    defaultMessage: `New`,
    description: `text for the "Release types" dropdown option "New"`,
  },
  help_whats_new_filter_select_option_improvement: {
    id: 'help.whats_new.filter_select_option_improvement',
    defaultMessage: `Improvement`,
    description: `text for the "Release types" dropdown option "Improvement"`,
  },
  help_whats_new_filter_select_option_fix: {
    id: 'help.whats_new.filter_select_option_fix',
    defaultMessage: `Fix`,
    description: `text for the "Release types" dropdown option "Fix"`,
  },
  help_whats_new_filter_select_option_removed: {
    id: 'help.whats_new.filter_select_option_removed',
    defaultMessage: `Removed`,
    description: `text for the "Release types" dropdown option "Removed"`,
  },
  help_whats_new_filter_select_option_experiment: {
    id: 'help.whats_new.filter_select_option_experiment',
    defaultMessage: `Experiment`,
    description: `text for the "Release types" dropdown option "Experiment"`,
  },
  help_whats_new_no_results: {
    id: 'help.whats.new_no_results',
    defaultMessage: `We can’t find any recent releases of that type.`,
    description: `text to display when the "What's new" API returns an empty array of articles"`,
  },
  help_whats_new_no_results_info: {
    id: 'help.whats.new_no_results_info',
    defaultMessage: `We only display three months of recent releases.`,
    description: `text to display when the "What's new" API returns an empty array of articles"`,
  },
  help_whats_new_no_results_clear_filter_button_label: {
    id: 'help.whats.new_no_results_clear_filter_button_label',
    defaultMessage: `Clear the filter`,
    description: `text to display when the "What's new" API returns an empty array of articles"`,
  },
  help_whats_new_no_results_clear_filter_info: {
    id: 'help.whats.new_no_results_clear_filter_button_label',
    defaultMessage: ` to try again.`,
    description: `text to display when the "What's new" API returns an empty array of articles"`,
  },
  help_whats_new_related_link_support: {
    id: 'help.whats_new.related_links.support',
    defaultMessage: `Support documentation`,
    description: `text for the "Support documentation" link in the related links section of "what's new" articles`,
  },
  help_whats_new_related_link_community: {
    id: 'help.whats_new.related_links.community',
    defaultMessage: `Discuss with community`,
    description: `text for the "Discuss with community" link in the related links section of "what's new" articles`,
  },
  help_show_more_button_label_more: {
    id: 'help.show_more_button.label_more',
    defaultMessage: 'Show {numberOfItemsLeft} more {itemsType}',
    description: '"Show more" text',
  },
  help_show_more_button_label_less: {
    id: 'help.show_more_button.label_less',
    defaultMessage: 'Show less',
    description: '"Show less" text',
  },
});
