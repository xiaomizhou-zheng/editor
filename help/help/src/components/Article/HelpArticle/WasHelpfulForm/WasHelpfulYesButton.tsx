import React from 'react';
import { injectIntl, WrappedComponentProps } from 'react-intl-next';
import {
  useAnalyticsEvents,
  UIAnalyticsEvent,
  AnalyticsContext,
} from '@atlaskit/analytics-next';
import Button from '@atlaskit/button/custom-theme-button';

import {
  name as packageName,
  version as packageVersion,
} from '../../../../version.json';
import { messages } from '../../../../messages';

const ANALYTICS_CONTEXT_DATA = {
  componentName: 'ArticleWasHelpfulYesButton',
  packageName,
  packageVersion,
};

interface Props {
  // Function executed when user clicks this button
  onClick?(
    event: React.MouseEvent<HTMLElement, MouseEvent>,
    analyticsEvent?: UIAnalyticsEvent,
  ): void;
  // Select state of the button
  isSelected?: boolean;
}

export const ArticleWasHelpfulYesButton: React.FC<
  Props & WrappedComponentProps
> = ({ isSelected = false, onClick, intl: { formatMessage } }) => {
  const { createAnalyticsEvent } = useAnalyticsEvents();

  const handleButtonClick = (
    event: React.MouseEvent<HTMLElement, MouseEvent>,
  ): void => {
    if (onClick) {
      const analyticsEvent: UIAnalyticsEvent = createAnalyticsEvent({
        action: 'clicked',
      });
      onClick(event, analyticsEvent);
    }
  };

  return (
    <Button onClick={handleButtonClick} isSelected={isSelected}>
      {formatMessage(messages.help_article_rating_option_yes)}
    </Button>
  );
};

const ArticleWasHelpfulYesButtonWithContext: React.FC<
  Props & WrappedComponentProps
> = (props) => {
  return (
    <AnalyticsContext data={ANALYTICS_CONTEXT_DATA}>
      <ArticleWasHelpfulYesButton {...props} />
    </AnalyticsContext>
  );
};

export default injectIntl(ArticleWasHelpfulYesButtonWithContext);
