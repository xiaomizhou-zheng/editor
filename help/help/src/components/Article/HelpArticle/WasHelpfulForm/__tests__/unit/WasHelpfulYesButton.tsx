import React from 'react';
import { createIntl, createIntlCache } from 'react-intl-next';
// eslint-disable-next-line import/no-extraneous-dependencies
import { render, fireEvent } from '@testing-library/react';

import { ArticleWasHelpfulYesButton } from '../../WasHelpfulYesButton';

import { messages } from '../../../../../../messages';
import AnalyticsListener from '@atlaskit/analytics-next/AnalyticsListener';

// Messages
const cache = createIntlCache();
const intl = createIntl(
  {
    locale: 'en',
    messages: {},
  },
  cache,
);
const messageYes = intl.formatMessage(messages.help_article_rating_option_yes);

const mockOnClick = jest.fn();
const analyticsSpy = jest.fn();

describe('ArticleContent', () => {
  it('Match snapshot', () => {
    const { container } = render(
      <ArticleWasHelpfulYesButton
        isSelected={false}
        onClick={mockOnClick}
        intl={intl}
      />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('props methods "onWasHelpfulYesButtonClick" and "onClick" should be executed when the user click the button', () => {
    const { getByText } = render(
      <AnalyticsListener channel="help" onEvent={analyticsSpy}>
        <ArticleWasHelpfulYesButton
          isSelected={false}
          onClick={mockOnClick}
          intl={intl}
        />
      </AnalyticsListener>,
    );

    const buttonYes = getByText(messageYes).closest('button');
    expect(buttonYes).not.toBeNull;

    if (buttonYes) {
      fireEvent.click(buttonYes);
      expect(mockOnClick).toHaveBeenCalledTimes(1);
    }
  });
});
