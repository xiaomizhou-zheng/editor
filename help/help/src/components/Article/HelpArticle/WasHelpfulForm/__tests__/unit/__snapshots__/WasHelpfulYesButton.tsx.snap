// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleContent Match snapshot 1`] = `


.emotion-1,
.emotion-1:hover,
.emotion-1:active,
.emotion-1:focus,
.emotion-1:visited,
.emotion-1:disabled,
.emotion-1[disabled] {
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  border-width: 0;
  border-radius: 3px;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: inherit;
  font-style: normal;
  font-family: inherit;
  font-weight: 500;
  max-width: 100%;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background 0.1s ease-out,box-shadow 0.15s cubic-bezier(0.47, 0.03, 0.49, 1.38);
  transition: background 0.1s ease-out,box-shadow 0.15s cubic-bezier(0.47, 0.03, 0.49, 1.38);
  white-space: nowrap;
  background: var(--ds-background-neutral, rgba(9, 30, 66, 0.04));
  color: var(--ds-text, #42526E)!important;
  cursor: pointer;
  height: 2.2857142857142856em;
  line-height: 2.2857142857142856em;
  padding: 0 10px;
  vertical-align: middle;
  width: auto;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  outline: none;
}

.emotion-1::-moz-focus-inner,
.emotion-1:hover::-moz-focus-inner,
.emotion-1:active::-moz-focus-inner,
.emotion-1:focus::-moz-focus-inner,
.emotion-1:visited::-moz-focus-inner,
.emotion-1:disabled::-moz-focus-inner,
.emotion-1[disabled]::-moz-focus-inner {
  border: 0;
  margin: 0;
  padding: 0;
}

.emotion-0 {
  opacity: 1;
  -webkit-transition: opacity 0.3s;
  transition: opacity 0.3s;
  margin: 0 2px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

<button
  class="emotion-1"
  tabindex="0"
  type="button"
>
  <span
    class="emotion-0"
  >
    Yes
  </span>
</button>
`;
